### PR TITLE
FIX: re-jigger deprecation of rcParams using machinery in __init__

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

This suppresses the warnings in the test suite and removes 

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant

### CODE REVIEW

**Comment by tacaswell** [comment]
Alternative to #10378 sorry, I did not see that one existed!  I think this one is a bit better as it cleans out all of the internal usage.

**Comment by jklymak** [comment]
No problem.  I was just hacking. Glad you have a proper solution.

File: lib/matplotlib/backends/qt_compat.py [hunk 1]
```diff
@@ -83,16 +83,16 @@
 if QT_API is None:
     # No ETS environment or incompatible so use rcParams.
     if rcParams['backend'] == 'Qt5Agg':
-        QT_API = rcParams['backend.qt5']
+        QT_API = QT_API_PYQT5
```
Comment (jklymak) [inline lib/matplotlib/backends/qt_compat.py]: OK, but this does't just deprecate the rcparam, it now ignores it, right?  Is that what you wanted?  Feel free to clear if so, but putting a red x so its noted..

Comment (tacaswell) [reply lib/matplotlib/backends/qt_compat.py]: Yes, but the rc deprecation tools do not have a gentler option

**Pull request merged**